scipy/stats/_variation.py

```diff
@@ -91,31 +94,37 @@ def variation(a, axis=0, nan_policy='propagate', ddof=0, *, keepdims=False):
     array([1.05109361, 0.31428986, 0.146483  ])
 
     """
+    print(axis)
+    print(a)
+    xp = array_namespace(a)
+    a = xp.asarray(a)
     # `nan_policy` and `keepdims` are handled by `_axis_nan_policy`
     n = a.shape[axis]
     NaN = _get_nan(a)
 
     if a.size == 0 or ddof > n:
         # Handle as a special case to avoid spurious warnings.
         # The return values, if any, are all nan.
-        shp = np.asarray(a.shape)
-        shp = np.delete(shp, axis)
-        result = np.full(shp, fill_value=NaN)
-        return result[()]
+        shp = xp.asarray(a.shape)
+        mask = xp.ones(shp.shape, dtype=xp.bool)
+        mask[axis] = False
```

Comment:
After looking at the JAX PR, I'll be looking out for code that mutates arrays when good alternatives are available. Here's one. This code can make `shp` a list instead of an array and `pop`ed. Not sure why array was used initially.

Reply:
Thanks, updated